src/main/java/dev/galacticraft/mod/content/entity/orbital/RocketEntity.java

```diff
@@ -290,12 +291,14 @@ public long getFuel() {
 
     @Override
     public void onBaseDestroyed() {
-
+        this.spawnAtLocation(GCItems.ROCKET.getDefaultInstance());
```

Comment:
This spawns a generic rocket item (rather than the type of rocket that was placed).
You'll need to use `::toNbt` to write the rocket part data to the ItemStack.

Reply:
Makes sense. I was actually thinking about that earlier today. I can probably address this tomorrow.